Feat: 방장 넘기기 기능

OHJINSEONG opened the pull request:
StudyRoomController에 방장 넘기기 api 추가
StudyRoomHostService에 자동 방장 넘기기 메소드를 autoTransferHost로 변경
StudyRoomHostService에 transferHost 메소드 추가

junhyeong00 committed:
Merge pull request #43 from junhyeong00/develop

Feat: 타이머 시간설정 기능 추가

junhyeong00 committed:
Ci: workflow branches 형식변경

junhyeong00 committed:
Merge pull request #44 from junhyeong00/github-action

Ci: workflow branches 형식변경

junhyeong00 committed:
Merge pull request #50 from junhyeong00/develop

뽀모도로 메이트 v1.0.0 배포

junhyeong00 committed:
Merge pull request #58 from Pomodoro-mate/develop

뽀모도로 메이트 v1.0.1 배포

OHJINSEONG committed:
Feat: 방장 넘기기 기능

    StudyRoomController에 방장 넘기기 api 추가
    StudyRoomHostService에 자동 방장 넘기기 메소드를 autoTransferHost로 변경
    StudyRoomHostService에 transferHost 메소드 추가

junhyeong00 committed:
Merge pull request #73 from Pomodoro-mate/develop

뽀모도로 메이트 v1.1.0 배포

junhyeong00 committed:
GenerationType.IDENTITY 설정

junhyeong00 committed:
GenerationType.IDENTITY 설정

junhyeong00 committed:
GenerationType.IDENTITY 제거

OHJINSEONG committed:
Fix: StudyRoomControllerTest에 테스트 의존성 추가

StudyRoomHostService를 @MockBean으로 추가

junhyeong00 commented on src/main/java/com/pomodoro/pomodoromate/studyRoom/controllers/StudyRoomController.java:
```diff
@@ -126,4 +119,16 @@ public ResponseEntity<Void> edit(
         return ResponseEntity.noContent().build();
     }
+
+    @Operation(summary = "방장 넘기기")
+    @PatchMapping("api/studyrooms/{studyRoomId}/host/{participantId}")
+    public ResponseEntity<Void> transferHost(
+            @RequestAttribute UserId userId,
+            @PathVariable Long studyRoomId,
+            @PathVariable Long participantId
+    ) {
+        studyRoomHostService.transferHost(userId, StudyRoomId.of(studyRoomId), ParticipantId.of(participantId));
+
+        return ResponseEntity.noContent().build();
+    }
```
방장이 바뀌면 다른 사람들한테도 바뀐걸 알려줘야할 것 같아요~

OHJINSEONG replied:
넵 웹소캣 연결 하겠습니다!

junhyeong00 commented on src/main/java/com/pomodoro/pomodoromate/studyRoom/applications/StudyRoomHostService.java:
```diff
@@ -23,7 +24,25 @@ public StudyRoomHostService(StudyRoomRepository studyRoomRepository,
 
     @Transactional
-    public void transferHost(StudyRoomId studyRoomId, ParticipantId hostId) {
+    public void transferHost(UserId userId, StudyRoomId studyRoomId, ParticipantId participantId) {
+        StudyRoom studyRoom = studyRoomRepository.findById(studyRoomId.value())
```
새로운 host를 participantId라고만 표현하면 어떤 의미인지 헷갈릴 것 같아요~

OHJINSEONG replied:
그렇네요 ㅎㅎ participantId -> newHostId 로 수정하였습니다

OHJINSEONG committed:
Merge branch 'main' of github.com:junhyeong00/Pomodoro-mate-backend into Feat/ojs-host_transfer

OHJINSEONG committed:
Chore: 방장 넘기기 기능 사용시 업데이트 추가

- StudyRoomHostService에 transferHost 메서드
-- 웹소캣 연결 추가 메서드 완료시 참가자 업데이트
-- 가독성을 높이기 위해 participantId -> newHostId 로 변경

OHJINSEONG committed:
Chore: 방장 넘기기 기능 웹소캣 엔드포인트 및 페이로드 변경

엔드포인트 : /sub/studyrooms/{studyRoomId}/participants -> /sub/studyrooms/{studyRoomId}/host

페이로드 : 참가자 리스트 -> 새로운 방장 ID

junhyeong00 commented on src/main/java/com/pomodoro/pomodoromate/studyRoom/applications/StudyRoomHostService.java:
```diff
@@ -49,11 +49,7 @@ public void transferHost(UserId userId, StudyRoomId studyRoomId, ParticipantId n
-
-        messagingTemplate.convertAndSend("/sub/studyrooms/" + studyRoomId + "/participants"
-                , participantSummariesDto);
+        messagingTemplate.convertAndSend("/sub/studyrooms/" + studyRoomId + "/host", newHostId);
```
흠 이 부분은 프론트랑 얘기를 해봐야할 것 같아요. 진행속도랑 복잡도 생각하면 participantSummariesDto 넘겨주는게 더 좋을 것 같긴합니다.

junhyeong00 reviewed (approved):
👍
